v1_1/crypto.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    crypto_box_keypair,
+    crypto_box_seal,
+    crypto_box_seal_open,
+    crypto_box_seed_keypair,
+    crypto_hash_sha256,
+    crypto_scalarmult,
+    crypto_scalarmult_base,
+    from_string,
+    IKeyPair,
+    randombytes_buf, compare,
+    to_string,
+    waitReady
+} from "../lib/sodium";
+
+export {waitReady}
+
+export interface ILocationData {
+    ent: Uint8Array;
+    pEnt: Uint8Array;
+    tr: tr;
+    pTr: pTr;
+}
+
+export interface tr {
+    sk: Uint8Array,
+    r2: Uint8Array
+}
+
+export interface pTr {
+    r1: Uint8Array,
+    r2: Uint8Array
+}
+
+export interface IVisit {
+    gR: Uint8Array;
+    h: Uint8Array;
+    c: Uint8Array;
+}
+
+/**
+ * This structure is as described in the chapter 4 of the white paper
+ *   CrowdNotifier: attack model, formalization and properties November 26, 2020
+ */
+export class CryptoV1_1 {
+    static setupHA(): IKeyPair {
+        return crypto_box_keypair();
+    }
+
+    static genCode(info: string): ILocationData {
```

Comment:
Here and everywhere: `info` should be a struct.

Reply:
Or be consistently converted into a byte array in `system.ts`

Reply:
info in crypto.ts -> Uint8array
info in system.ts -> LocationInfo protobuf